### PR TITLE
Upgrade pytest

### PR DESCRIPTION
**Pull request by wagnerand**

### CODE REVIEW

**Comment by wagnerand** [comment]
All test changes are to get rid of deprecated `yield`statements.

File: tests/test_content.py [hunk 1]
```diff
@@ -242,23 +243,22 @@ def test_css():
...
 
 
-def test_hidden_files():
+@pytest.mark.parametrize("test_input", [
+    {'.hidden': 'tests/resources/content/junk.xpi'},
+    {'dir/__MACOSX/foo': 'tests/resources/content/junk.xpi'},
+    {'dir/.foo.swp': 'tests/resources/content/junk.xpi'},
+    {'dir/file.old': 'tests/resources/content/junk.xpi'},
+    {'dir/file.xul~': 'tests/resources/content/junk.xpi'}
+])
```
Comment (EnTeQuAk) [inline tests/test_content.py]: :heart:

File: tests/test_content.py [hunk 2]
```diff
@@ -242,23 +243,22 @@ def test_css():
...
-                      {'dir/__MACOSX/foo': 'tests/resources/content/junk.xpi'},
-                      {'dir/.foo.swp': 'tests/resources/content/junk.xpi'},
-                      {'dir/file.old': 'tests/resources/content/junk.xpi'},
-                      {'dir/file.xul~': 'tests/resources/content/junk.xpi'}):
-        yield test_structure, structure
+    err = ErrorBundle()
+    err.supported_versions = {}
+    mock_package = MockXPI(test_input)
+    content.test_packed_packages(err, mock_package)
+    print err.print_summary(verbose=True)
```
Comment (EnTeQuAk) [inline tests/test_content.py]: deug print?

**Comment by EnTeQuAk** [comment]
I love parametrize, r+wc

**Comment by wagnerand** [comment]
Let's clean up `print` statements in a separate PR as the tests are full of them.

**Comment by wagnerand** [comment]
Thank you!

**Pull request merged**